extensions/arc/deployment/src/test/java/io/quarkus/arc/test/config/ConfigMappingTest.java

```diff
@@ -252,6 +256,36 @@ void maps() {
         assertEquals(8080, maps.group().get("server").port());
     }
 
+    public interface ServerBase {
+        Map<String, String> server();
+    }
+
+    @ConfigMapping(prefix = "maps.base")
+    public interface MapsWithBase extends ServerBase {
+        @Override
+        Map<String, String> server();
+
+        Map<String, Server> group();
+
+        interface Server {
+            String host();
+
+            int port();
+        }
+    }
+
+    @Inject
+    MapsWithBase mapsWithBase;
+
+    @Test
+    void mapsWithBase() {
+        assertEquals("localhost", mapsWithBase.server().get("host"));
+        assertEquals(8080, Integer.valueOf(mapsWithBase.server().get("port")));
```

Comment:
```suggestion
        assertEquals(8080, Integer.parseInt(mapsWithBase.server().get("port")));
```

Reply:
`Integer.valueOf` calls `Integer.parseInt` internally

Reply:
It's just to remove the unecessary autoboxing, since you're comparing primitives here, but it's not that relevant anyway